### PR TITLE
Add WSD (Warmup-Stable-Decay) scheduler for SFT

### PR DESCRIPTION
**Pull request by xingyaoww**
# Add WSD (Warmup-Stable-Decay) Learning Rate Scheduler

## Overview
This PR adds a new learning rate scheduler called WSD (Warmup-Stable-Decay) that provides more control over the learning rate schedule during training. The WSD scheduler extends the traditional cosine scheduler by adding a stable phase where the learning rate remains constant.

## Features
- **Three-phase schedule**: Warmup → Stable → Decay
- **Configurable stable phase**: Control what percentage of training maintains a constant learning rate
- **Compatible with existing code**: Minimal changes to the trainer infrastructure
- **Default to cosine**: Maintains backward compatibility with existing configurations

## Implementation Details
1. Added `get_wsd_schedule_with_warmup` function to `verl/utils/torch_functional.py`
2. Updated the SFT trainer to support the new scheduler type
3. Added `lr_scheduler: cosine` as the default in the SFT trainer config

Here's the reference implementation: https://github.com/kozistr/pytorch_optimizer/blob/6397d56279ad80b26c4bba7fb4b04852b517fdeb/pytorch_optimizer/lr_scheduler/wsd.py#L8

## Usage
To use the WSD scheduler, set the following in your configuration:
```yaml
optim:
  lr_scheduler: wsd  # Options: 'cosine' (default) or 'wsd'
```

## Benefits
- Better control over learning rate behavior during training
- Potentially improved training stability for certain tasks
- Allows experimentation with different learning rate schedules without code changes

(trying to get this in to make sure my own branch don't end up with huge chunk of git conflict 😓 )

### CODE REVIEW

File: verl/trainer/config/sft_trainer.yaml [hunk 1]
```diff
@@ -38,6 +38,7 @@ optim:
   weight_decay: 0.01
   warmup_steps_ratio: 0.1
   clip_grad: 1.0
```
Comment (eric-haibin-lin) [inline verl/trainer/config/sft_trainer.yaml:40]: could you update config.rst with the new added option? thx

Comment (xingyaoww) [reply verl/trainer/config/sft_trainer.yaml:40]: Does this works?

https://github.com/volcengine/verl/pull/1041/commits/725289e6377ecefcc1aff6811b585fa0c4f60b9e

**Review by eric-haibin-lin — DISMISSED** [review]
I fixed a rendering issue. kicking off the CI now

**Comment by eric-haibin-lin** [comment]
could you run `scripts/format.sh`

**Comment by xingyaoww** [comment]
@eric-haibin-lin done!

**Review by eric-haibin-lin — APPROVED** [review]
